printf/src/arg.rs

```diff
@@ -0,0 +1,239 @@
+use super::printf_impl::Error;
+use std::result::Result;
+use widestring::{Utf32Str as wstr, Utf32String as WString};
+
+/// Printf argument types.
+///  Note no implementation of ToArg constructs the owned variants (String and WString);
+/// callers can do so explicitly.
+#[derive(Debug, PartialEq)]
+pub enum Arg<'a> {
```

Comment:
Very minor nit, but can we keep the code structured the same way but make this `pub(crate)` and re-export it in `printf/lib.rs` as `pub using arg::Arg`? (Same with the traits)

I don't see the point in downstream users having to do `use printf::sprintf;` and `use printf::arg::Arg` separately from separate modules when you need `Arg` anyway.

Reply:
Turns out that if a type is `pub(crate)` it can't be re-exported as `pub` (Rust modules continue to confound) but your suggestion of adding `pub use` to lib.rs was good, I adopted it.